### ISSUE TITLE
Changeset apply function ignores non-serial primary field during insertion

### ISSUE THREAD

**Issue by talkanbaev-artur** [go-rel/changeset#38]
Hi! 
It seems that changeset struct only accepts the serial primary keys. In case of setting a custom primary key, it never gets inserted:
https://github.com/go-rel/changeset/blob/c1cf3574d23f2bf8d3162b374f142df804672017/changeset.go#L90

To reproduce:
1) create the table with string primary key
2) create changeset which in case of primary key, generates its value
3) try to insert the new record with changeset mutator

Expected behaviour:
For non-serial primary keys and insertion operation, primary filed should not be ignored and applied as a mutation.

For me just removing this check worked, however, I think that a more sophisticated logic should be applied (the value should not be mutated during update and serial primary key should not applied same way).

**Comment by Fs02** [comment]
another workaround is to use both rel.Set and Changeset, ex:

```go
repo.Insert(context.TODO(), &product, ch, rel.Set("id", "..."))
```

the reason for that condition is as a safe guard from accidentally updating id, but I need to take a look at this again and maybe comes with better fix
(if you have any better suggestion or fix, feel free to let me know 😄 )

**Comment by talkanbaev-artur** [comment]
Hmmmm... I see.

I have made my own *almost safe* workaround: checked if the field is not serial (not int) and was mutated in the changeset.
I think if the field was mutated in the changeset explicitly, then, probably, it is not accident. 

To use `rel.Set` would mean to extract the value from the changeset for me.
```go
if ch.Values()["uuid"].(UUID) == "" {
	newUser := NewUser(ch.Get("email").(string))
	changeset.PutChange(ch, "uuid", newUser.UUID)
        //other code
}
````
So the whole generation process is tied to the changeset. Anyway, it is just my case and I have not discovered all possibilities in this code yet.

**Comment by Fs02** [comment]
taking a look again, looks like accidental update of id can be prevented, because id is whitelisted manually in this params, so attacker specifed id will be ignored if Cast function is used 🤔 


```go
changeset.Cast(product, params.ParseJSON(`{"name":  "soap", "price": 2000}`, []string{"name", "price"})
```

**Comment by talkanbaev-artur** [comment]
Yes the PK should never be casted, and only changed by the internal logic of the changeset. Same applies for password hashes and other sensitive data. In my case the branch with manual PK assign happens only when there is PK at all - when record is new.

I understand that this decision would put additional responsibility on the developer, so the choice is up to you.

**Comment by talkanbaev-artur** [comment]
> but we also need to update Convert function to remove id from changes and place it maybe in values 
I'm sorry, but I do not get it. To me it seems, that PK can be explicitly changed only from some kind of `PutChange` method. 
So, I will try to remove the id from changes in Convert. However, imo, it should be ignored in that method, and not to be put in values.

**Comment by talkanbaev-artur** [comment]
I found another way to create the changeset from the method `Change`. So I created a flag and now the only way to mutate PK is to have a changeset created from `Cast` with explicitly set PK or to execute `PutChange` on the changeset, which was created from `Cast`.

Sorry for over complicating the logic. Probably this solution is rather stupid and creates holes in the app. I see a rather more elegant solution in having this flag set to be true by default, and setting it to false only if `PutChange` is called explicitly for the PK. Let me know, if I should continue or it is better to leave code as it is.
